plugins/out_s3/s3.c

```diff
@@ -657,9 +688,143 @@ static int cb_s3_init(struct flb_output_instance *ins,
                           "use_put_object must be enabled when Apache Arrow is enabled");
             return -1;
         }
+        if (ctx->use_put_object == FLB_FALSE && ctx->compression == FLB_AWS_COMPRESS_PARQUET) {
+            flb_plg_error(ctx->ins,
+                          "use_put_object must be enabled when parquet is enabled");
+            return -1;
+        }
         ctx->compression = ret;
     }
 
+    /* Parquet */
+    ctx->parquet_compression = NULL;
+    ctx->parquet_record_type = NULL;
+    ctx->parquet_schema_type = NULL;
+    ctx->parquet_schema_file = NULL;
+
+    if (ctx->compression == FLB_AWS_COMPRESS_PARQUET) {
```

Comment:
Please also validate that the new parquet fields are cleaned up when ctx is cleaned